bugsnag-android-core/src/main/java/com/bugsnag/android/Breadcrumb.java

```diff
@@ -68,7 +66,7 @@ public BreadcrumbType getType() {
 
     @NonNull
     public Map<String, String> getMetadata() {
-        return metadata;
+        return Collections.unmodifiableMap(metadata);
```

Comment:
Aren't there cases where people might be changing/removing breadcrumb metadata in a callback? The case which comes to mind is intent extras from the `EventReceiver`.

Reply:
While it's technically possible people are using the API in this way, I don't think we'd be breaking the contract as we haven't specified anything about what the Map is that's returned. Has the intent extras/other examples come up regularly before?

Also stepping back a bit, I think ideally `Breadcrumb` should return a `MetaData` here as it would be more consistent with the rest of our API. Although this would be a breaking change that should be addressed separately.

Reply:
> Has the intent extras/other examples come up regularly before?

I don't know about regularly, but there have been support requests in the past about removing metadata from automatic breadcrumbs. Since reports from the NDK pass through `beforeSend` callbacks, the breadcrumbs for native reports could also be reasonably customized. 

Making this collection unmodifiable doesn't make breadcrumb usage safer since the interface to leaving a breadcrumb doesn't expose the breadcrumb instance.

Reply:
I've altered this to return the backing map directly for now, and will raise a ticket to improve the API in future.